pkg/render/policyrecommendation.go

```diff
@@ -0,0 +1,329 @@
+// Copyright (c) 2023 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package render
+
+import (
+	v3 "github.com/tigera/api/pkg/apis/projectcalico/v3"
+	"github.com/tigera/operator/pkg/ptr"
+	relasticsearch "github.com/tigera/operator/pkg/render/common/elasticsearch"
+	"github.com/tigera/operator/pkg/render/common/networkpolicy"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+	"github.com/tigera/operator/pkg/render/common/securitycontext"
+	"github.com/tigera/operator/pkg/tls/certificatemanagement"
+)
+
+// The names of the components related to the PolicyRecommendation APIs related rendered objects.
+const (
+	ElasticsearchPolicyRecommendationUserSecret = "tigera-ee-policy-recommendation-elasticsearch-access"
+
+	PolicyRecommendationName = "tigera-policy-recommendation"
+
+	PolicyRecommendationClusterRoleName        = PolicyRecommendationName
+	PolicyRecommendationClusterRoleBindingName = PolicyRecommendationName
+	PolicyRecommendationDeploymentName         = PolicyRecommendationName
+	PolicyRecommendationNamespace              = PolicyRecommendationName
+	PolicyRecommendationServiceName            = "policy-recommendation"
+	PolicyRecommendationServiceAccountName     = PolicyRecommendationName
+	PolicyRecommendationPolicyName             = networkpolicy.TigeraComponentPolicyPrefix + PolicyRecommendationName
+
+	PolicyRecommendationControllerName = "policy-recommendation-controller"
```

Comment:
This shouldn't be defined in render - it's just the name of the operator controller and isn't used in the render package at all.

The render package isn't meant to hold all of the global variables, just those needed for rendering k8s resources.